lightning/src/ln/channel.rs

```diff
@@ -5298,6 +5625,58 @@ impl<SP: Deref> Channel<SP> where
 		}
 	}
 
+	pub fn tx_signatures(&mut self, msg: &msgs::TxSignatures) -> Result<(Option<msgs::TxSignatures>, Option<Transaction>), ChannelError> {
+		if let Some(ref mut signing_session) = self.interactive_tx_signing_session {
```

Comment:
We should check state and wait for persistence of initial `ChannelMonitor` here, but in `InteractiveTxSigningSession` we will only allow our `tx_signatures` to be sent if we have received a `commitment_signed`, but looks like something went missing in `commitment_signed_initial_v2` along the way.

Nothing broke in the test as we always send an empty witnesses array in `tx_signatures` since we don't support contributing funds/inputs here. I'll make sure it goes through the full flow in this PR anyway so we can catch that error.

Thanks!